integration-tests/redis_test.go

```diff
@@ -78,10 +78,139 @@ var _ = Describe("Redis", Label("Redis"), func() {
+		DescribeTable("should check property constraints",
+			func(params map[string]any, expectedErrorMsg string) {
+				_, err := broker.Provision(redisServiceName, "small", params)
+				Expect(err).To(MatchError(ContainSubstring(expectedErrorMsg)))
+			},
+			Entry(
+				"invalid region",
+				map[string]any{"region": "-Asia-northeast1"},
+				"region: Does not match pattern '^[a-z][a-z0-9-]+$'",
+			),
+			Entry(
+				"instance name minimum length is 6 characters",
+				map[string]any{"instance_name": stringOfLen(5)},
+				"instance_name: String length must be greater than or equal to 6",
+			),
+			Entry(
+				"instance name maximum length is 40 characters",
+				map[string]any{"instance_name": stringOfLen(99)},
+				"instance_name: String length must be less than or equal to 40",
+			),
+			Entry(
+				"instance name invalid characters",
+				map[string]any{"instance_name": ".aaaaa"},
+				"instance_name: Does not match pattern '^[a-z][a-z0-9-]+$'",
+			),
+		)
+
+		DescribeTable(
+			"should prevent modifying `plan defined properties`",
+			func(prop string, value any) {
+				_, err := broker.Provision(redisServiceName, "small", map[string]any{prop: value})
+
+				Expect(err).To(MatchError(
+					ContainSubstring(
+						"plan defined properties cannot be changed",
+					),
+				))
+
+				Expect(mockTerraform.ApplyInvocations()).To(HaveLen(0))
+			},
+			Entry("cache_size", "cache_size", 9),
+			Entry("node_count", "node_count", 5),
+			Entry("redis_version", "redis_version", "4.0"),
+		)
 
-			Expect(err).To(MatchError(ContainSubstring("region: Does not match pattern '^[a-z][a-z0-9-]+$'")))
+		DescribeTable(
+			"should disallow `user_input` properties with the same name as some `computed_input` for clarity",
+			func(prop string, value any) {
+				_, err := broker.Provision(redisServiceName, "small", map[string]any{prop: value})
+
+				Expect(err).To(MatchError(
+					ContainSubstring(
+						"additional properties are not allowed",
+					),
+				))
+
+				Expect(mockTerraform.ApplyInvocations()).To(HaveLen(0))
+			},
+			Entry("labels", "labels", "a-valid-list-of-labels"),
+		)
+
+		DescribeTable("currently doesn't provide validations constraints for some properties",
+			func(params map[string]any) {
+				_, err := broker.Provision(redisServiceName, "small", params)
+				Expect(err).NotTo(HaveOccurred())
+			},
+			Entry(
+				"aws_vpc_id is never validated by the brokerpak logic",
+				map[string]any{"aws_vpc_id": stringOfLen(99)},
+			),
+			Entry(
+				"node_type is never validated by the brokerpak logic",
+				map[string]any{"node_type": stringOfLen(99)},
+			),
+			Entry(
+				"elasticache_subnet_group currently is never validated by the brokerpak logic",
+				map[string]any{"elasticache_subnet_group": stringOfLen(99)},
+			),
+			Entry(
+				"elasticache_vpc_security_group_ids is never validated by the brokerpak logic",
+				map[string]any{"elasticache_vpc_security_group_ids": stringOfLen(99)},
+			),
+		)
```

Comment:
thoughts: sometimes the Terraform provider applies some validations but those are not reflected in this layer of tests. I am not sure if this test adds some extra value

Reply:
Agreed. I will remove this, just wanted to raise awareness in case this passed unadvertised.